libs/template/resolver.go

```diff
@@ -0,0 +1,97 @@
+package template
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/databricks/cli/libs/git"
+)
+
+type Resolver struct {
+	// One of the following three:
+	// 1. Path to a local template directory.
+	// 2. URL to a Git repository containing a template.
+	// 3. Name of a built-in template.
+	TemplatePathOrUrl string
+
+	// Path to a JSON file containing the configuration values to be used for
+	// template initialization.
+	ConfigFile string
+
+	// Directory to write the initialized template to.
+	OutputDir string
+
+	// Directory path within a Git repository containing the template.
+	TemplateDir string
+
+	// Git tag or branch to download the template from. Only one of these can be
+	// specified.
+	Tag    string
+	Branch string
+}
+
+var ErrCustomSelected = errors.New("custom template selected")
+
+// Configures the reader and the writer for template and returns
+// a handle to the template.
+// Prompts the user if needed.
+func (r Resolver) Resolve(ctx context.Context) (*Template, error) {
+	if r.Tag != "" && r.Branch != "" {
+		return nil, fmt.Errorf("only one of --tag or --branch can be specified")
```

Comment:
Refers to flag names. Will go out of sync with the command.

Please mark these flags as mutually exclusive in the command itself, then the error here can refer to the concepts and not the flag names.

Reply:
Marking the flags exclusive has an ugly error message. Reverted back to the original check instead.

```
.venv➜  cli git:(refactor-bundle-init-squashed) ✗ cli bundle init --branch b --tag t
Using workspace profile: dogfood
Error: if any flags in the group [tag branch] are set none of the others can be; [branch tag] were all set
```